src/Compilers/Core/Portable/Operations/OperationExtensions.cs

```diff
@@ -14,6 +14,14 @@ namespace Microsoft.CodeAnalysis.Operations
 {
     public static partial class OperationExtensions
     {
+        /// <summary>
+        /// Helper function to simplify the access to the function pointer signature of an FunctionPointerInvocationOperation
+        /// </summary>
+        public static IMethodSymbol? GetFunctionPointerSignature(this IFunctionPointerInvocationOperation functionPointer)
```

Comment:
>IMethodSymbol? [](http://example.com/codeflow?start=22&length=14)

The fact that this API can return null feels somewhat unexpected. Under what circumstances can this happen?

Reply:
With ROSLYN_TEST_IOPERATION enabled the test "StackallocOfFunctionPointers" failed on that.
This test has a pointer to function pointers which is accessed via indexer -> `p[i]`.
But you are right it a bug somewhere else the `Type` should be not null but the function pointer.

Reply:
I changed it and adjusted the NoneOperation creation...